### DISCUSSION TITLE
How to attach debugger from VSCode

### DISCUSSION THREAD

**Discussion by JulienRst** [jmcdo29/nest-commander#583] [Q&A]
Hello everybody 😄 

I'm working on a small project with nest-commander and I'm in dire need for help.

How do you make the debugger work when running a command like "yarn myProject foo" 

Is this even possible ? 

Here's my debugger config for now 

```
{
      "name": "Attach to app",
      "port": 9229,
      "request": "attach",
      "skipFiles": ["<node_internals>/**"],
      "type": "node"
}
``` 

I think it may be because the VSCode debugger is looking to attach to a server and not a "script" so I'm a little bit lost there.

Any help ?

**Comment by sonic1981** [comment]
This is what I did:

```json
 {
      "type": "node",
      "request": "launch",
      "name": "Debug command",
      "runtimeArgs": [
        "--inspect", "node_modules/.bin/ts-node", "apps/auth0-scripts/src/main.ts", "command-name"
      ],
      "sourceMaps": true,
      "cwd": "${workspaceRoot}",
      "console": "integratedTerminal",
      "protocol": "inspector",
      "skipFiles": [
        "<node_internals>/**"
      ],
    }
 ```
    
You'll need to install ts-node to use this method `npm install ts-node -D`

I also had to add source maps into webpack.config.js

```typescript
const path = require('path')

module.exports = (options) => {
  const config = { ...options }
  config.devtool = 'inline-source-map'
  return config
}
```
